lib/graphql/account_settings/account_settings_db.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:autonomy_flutter/graphql/account_settings/account_settings_client.dart';
+
+abstract class AccountSettingsDB {
+  Future<void> download({List<String>? keys});
+
+  Future<void> forceUpload();
+
+  List<Map<String, String>> query(List<String> keys);
+
+  Future<void> write(List<Map<String, String>> settings);
+
+  Future<void> delete(List<String> keys);
+
+  Map<String, dynamic> get caches;
+
+  Future<bool> didMigrate();
+
+  Future<void> setMigrated();
+
+  String getFullKey(String key);
+
+  String get migrateKey;
+
+  String get prefix;
+
+  void clearCache();
+}
+
+class AccountSettingsDBImpl implements AccountSettingsDB {
+  final AccountSettingsClient _client;
+  final String _prefix;
+
+  AccountSettingsDBImpl(this._client, this._prefix);
+
+  final Map<String, String> _caches = {};
+
+  static const String _migrateKey = 'didMigrate';
+
+  @override
+  Future<void> download({List<String>? keys}) async {
+    late List<Map<String, String>> values;
+    if (keys != null) {
+      values =
+          await _client.query(vars: {'keys': keys.map(getFullKey).toList()});
+    } else {
+      values = await _client.query(vars: {'search': '$_prefix.'});
+    }
+    for (var value in values) {
+      if (value['key'] == null || value['value'] == null) {
+        continue;
+      }
+      _caches[_removePrefix(value['key']!)] = value['value']!;
+    }
+  }
+
+  @override
+  Future<void> forceUpload() async {
+    final List<Map<String, String>> data = [];
+    _caches.forEach((key, value) {
+      data.add({'key': getFullKey(key), 'value': value});
+    });
+    await write(data);
+  }
+
+  @override
+  List<Map<String, String>> query(List<String> keys) => keys
+      .map((key) => {'key': key, 'value': _caches[key]})
+      .where((element) => element['value'] != null)
+      .map((e) => {'key': e['key']!, 'value': e['value']!})
+      .toList();
+
+  @override
+  Future<void> write(List<Map<String, String>> settings) async {
+    settings.removeWhere(
+        (element) => element['key'] == null || element['value'] == null);
+    final settingsFullKeys = settings
+        .map((e) => {'key': getFullKey(e['key']!), 'value': e['value']!})
+        .toList();
+    final isSuccess = await _client.write(data: settingsFullKeys);
```

Comment:
Do we need update to server every time we update in cache?

Reply:
yes, because we don't save it in local. if you add an address, kill app, reopen, and that address does not appear, that would be a bug. beside, we only save thing that barely change like address, connection, settings